--- a/daed/Makefile
+++ b/daed/Makefile
@@ -7,7 +7,7 @@ include $(TOPDIR)/rules.mk
 PKG_NAME:=daed
 PKG_VERSION:=2024.11.27
 DAED_VERSION:=daed-b44c472
-WING_VERSION:=wing-c4c7159
+WING_VERSION:=wing-93d987c
 CORE_VERSION:=core-aea92a0
 PKG_RELEASE:=1
 
@@ -70,14 +70,13 @@ endef
 define Build/Prepare
 	( \
 		$(TAR) --strip-components=1 -C $(PKG_BUILD_DIR)/../ -xzf $(DL_DIR)/$(PKG_NAME)-$(PKG_VERSION).tar.gz ; \
-		rm -rf $(PKG_BUILD_DIR) && git clone https://github.com/daeuniverse/dae-wing $(BUILD_DIR)/$(PKG_NAME)-$(PKG_VERSION)/wing && git -C $(BUILD_DIR)/$(PKG_NAME)-$(PKG_VERSION)/wing checkout ${WING_VERSION: -7} ; \
+		rm -rf $(PKG_BUILD_DIR) && git clone https://github.com/sdandroid/dae-wing $(BUILD_DIR)/$(PKG_NAME)-$(PKG_VERSION)/wing && git -C $(BUILD_DIR)/$(PKG_NAME)-$(PKG_VERSION)/wing checkout ${WING_VERSION: -7} ; \
 		rm -rf $(PKG_BUILD_DIR)/dae-core && git clone https://github.com/daeuniverse/dae $(PKG_BUILD_DIR)/dae-core && git -C $(PKG_BUILD_DIR)/dae-core checkout ${CORE_VERSION: -7} ; \
 		pushd $(PKG_BUILD_DIR)/dae-core ; \
 		git submodule update --init ; \
 		go mod tidy ; \
 		popd ; \
 		pushd $(BUILD_DIR)/$(PKG_NAME)-$(PKG_VERSION)/wing ; \
-		wget -qO - https://github.com/daeuniverse/dae-wing/commit/ca02961.patch | patch -p1 ; \
 		go mod tidy ; \
 		popd ; \
 		pushd $(BUILD_DIR)/$(PKG_NAME)-$(PKG_VERSION) ; \
--- /dev/null
+++ b/libcron/Makefile
@@ -0,0 +1,47 @@
+
+# SPDX-License-Identifier: GPL-3.0-only
+#
+# Copyright (C) 2021 ImmortalWrt.org
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=libcron
+PKG_RELEASE:=1
+
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_URL:=https://github.com/PerMalmberg/libcron.git
+PKG_SOURCE_DATE:=2023-11-14
+PKG_SOURCE_VERSION:=41f238ceb09d4179e7346d78584a0c978e5d0059
+PKG_MIRROR_HASH:=c7b0651566153b1d641e3b5ece50474c8556d42345779f19a5f22814a8183c38
+
+PKG_LICENSE:=MIT
+PKG_LICENSE_FILES:=LICENSE
+PKG_MAINTAINER:=Tianling Shen <cnsztl@immortalwrt.org>
+
+PKG_BUILD_PARALLEL:=1
+CMAKE_INSTALL:=1
+
+include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/cmake.mk
+
+define Package/libcron
+  SECTION:=lib
+  CATEGORY:=Libraries
+  URL:=https://github.com/PerMalmberg/libcron
+  TITLE:=A C++ scheduling library using cron formatting
+  DEPENDS:=+libstdcpp
+endef
+
+define Package/libcron/description
+  Libcron offers an easy to use API to add callbacks with corresponding
+  cron-formatted strings.
+endef
+
+CMAKE_OPTIONS+= -DBUILD_SHARED_LIBS=ON
+
+define Package/libcron/install
+	$(INSTALL_DIR) $(1)/usr/lib/
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib/liblibcron.so $(1)/usr/lib/
+endef
+
+$(eval $(call BuildPackage,libcron))
--- a/luci-app-daed/root/etc/daed/daed_sub.sh
+++ b/luci-app-daed/root/etc/daed/daed_sub.sh
@@ -23,7 +23,7 @@ update_subscription() {
 }
 
 reload() {
-	curl -X POST -H "Authorization: $TOKEN" -d '{"query":"mutation Run($dry: Boolean!) {\n  run(dry: $dry)\n}","variables":{"dry":false},"operationName":"Run"}' $GRAPHQL_URL
+	curl -X POST -H "Authorization: $TOKEN" -d '{"query":"mutation RunForSubscription($dry: Boolean!) {\n  runForSubscription(dry: $dry)\n}","variables":{"dry":false},"operationName":"RunForSubscription"}' $GRAPHQL_URL
 }
 
 resetcron() {
@@ -33,4 +33,4 @@ resetcron() {
   crontab $CRON_FILE
 }
 
-login && update_subscription && reload && resetcron
+login && update_subscription && reload
--- a/luci-app-daed/root/etc/init.d/luci_daed
+++ b/luci-app-daed/root/etc/init.d/luci_daed
@@ -14,7 +14,7 @@ RANDOM_NUM=$((RANDOM_SEED % 10 + 1))
 setcron() {
   touch $CRON_FILE
   sed -i '/daed_sub.sh/d' $CRON_FILE 2>/dev/null
-  [ "$(uci -q get daed.config.subscribe_auto_update)" -eq 1 ] && echo "${RANDOM_NUM} $(uci -q get daed.config.subscribe_update_day_time) * * $(uci -q get daed.config.subscribe_update_week_time) /etc/daed/daed_sub.sh >/dev/null 2>&1" >>$CRON_FILE
+  [ "$(uci -q get daed.config.subscribe_auto_update)" -eq 1 ] && echo "*/1 * * * * /etc/daed/daed_sub.sh >/dev/null 2>&1" >>$CRON_FILE
   crontab $CRON_FILE
 }
 
